src/hooks/useFocusTrap.ts

```diff
@@ -0,0 +1,46 @@
+// Based on https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element
+import { useRef, useEffect } from 'react';
+
+const KEYCODE_TAB = 9;
```

Comment:
Should use https://github.com/EightfoldAI/octuple/blob/main/src/shared/utilities/eventKeys.ts